### PR TITLE
refactor(modal): migrate more animations to animateCss

### PR DESCRIPTION
**Pull request by Foxandxss**
Now that we are breaking stuff, lets clean the backdrop animation a bit.

We should create a milestone too for doing better animations tho.

### CODE REVIEW

File: src/modal/modal.js [hunk 1]
```diff
@@ -76,24 +70,16 @@ angular.module('ui.bootstrap.modal', ['ui.bootstrap.stackedMap'])
...
     function linkFn(scope, element, attrs) {
       if (attrs.modalInClass) {
-        if ($animateCss) {
-          $animateCss(element, {
-            addClass: attrs.modalInClass
-          }).start();
-        } else {
-          $animate.addClass(element, attrs.modalInClass);
-        }
+        $animateCss(element, {
```
Comment (wesleycho) [inline src/modal/modal.js]: Wrap this in a function to remove duplication

Comment (Foxandxss) [inline src/modal/modal.js]: where?

Comment (wesleycho) [inline src/modal/modal.js]: function animateIn(cssKey) {
  var animateObj = {};
  animateObj[cssKey] = attes.modalInClass;
  return $animateCss(element, animateObj).start();
}

Or something like that

Comment (Foxandxss) [inline src/modal/modal.js]: Yeah, the problem is that several directives need it and I am not sure I want another service just for this.

Comment (wesleycho) [inline src/modal/modal.js]: I just meant a helper function just for this directive.

Comment (Foxandxss) [inline src/modal/modal.js]: But the next one has the same need.

Comment (wesleycho) [inline src/modal/modal.js]: I guess I'm fine with that, but it looks a little ugly to have that duplication. This can just be merged in then.

**Pull request closed without merging**